### DISCUSSION TITLE
Color profiles on .png

### DISCUSSION THREAD

**Discussion by w4rner** [ImageMagick/ImageMagick#3732] [Digital Image Processing]
Hey guys,

Having the sRGB (sRGB IEC61966-2.1) color profile on .png makes them appear different colour when published on a webpage. Please see my [blog post](https://w4rner-test.micro.blog/2021/05/15/help-why-are.html) for more info & examples.

a). if I wanted to remove the color profile from a .png would [this imagemagick method work](https://www.rigacci.org/wiki/doku.php/doc/appunti/software/imagemagick_color_management) on .png as well as .tiff:
`convert image.tiff -strip image-no-profile.tiff`

b). (bigger picture question for you image heads) is it even good practise to strip this color profile, given that it is considered the ? Is the problem I'm having more fundamental that I'm writing the wrong colour to my .png images in the first place?
From ImageMagick website:
"most image formats use SRGB as the default working color space"
is 'color space' there mean the same thing as 'colour profile'? If so, then surely I should be leaving sRGB colour profile on my image.

Thanks for help & advice to this newbie!

W4RNER

**Comment by snibgo** [comment]
Your web page refers to "Adobe RGB", but I don't know what that means. Perhaps it means "Adobe RGB (1998)", which is a colorspace that has a larger gamut than sRGB.

"-strip" will remove any embedded ICC profile, so most software will then assume the image is sRGB. If the embedded profile was "Adobe RGB (1998)" then the result will look different, less saturated.

If the embedded profile was sRGB then the image may look slightly different, because there are many versions of sRGB with small differences.

Not all display software will take notice of embedded profiles.

If your image has an embedded ICC profile and the exact appearance is important, you should not remove the profile.

"Colorspace" (or "colour space" etc) means the definition of what colours should be shown for given pixel values. In particular, it includes colour primaries and transfer curves.

An ICC colour profile is a particular format for colorspace metadata.

**Comment by w4rner** [comment]
There’s a few different things going on here:

1) Intrinsic properties of images:
I hear you that colorspace is different from color (sync) profile.
What is an ICC profile?
When you say ‘display software’, is that the software Carrd use to render the website? Why are they consistently making it look darker?

2). My attempts to measure the differences in how colours appear, using Digital Color Meter
“Adobe RGB” on the blog post is one the measures I look at, along with ’sRGB’ and ‘Display Native Values”.
I was trying to ascertain if they colours were ‘different’ in some ground truth sense or if it was just displaying different in different settings.
It definitely seems that the ones with a Color Sync Profile consistently changes the color.

Side-note of interest: I was super curious that Firefox + iPad Sidecar = Brave on Mac
And the fact that doing two seemingly unrelated things (1. displaying on Brave instead of Firefox and 2. adding a sRGB color sync profile to the intrinsic image) had an identical effect across measures!

Anything you can help me clarify here will be appreciated.
Remember: my aim is to consistently output a specific colour (cerulean #007ba6) to appear as identical as possible across settings.

W4RNER

**Comment by snibgo** [comment]
> Remember: my aim is to consistently output a specific colour (cerulean #007ba6) to appear as identical as possible across settings.

To do that, you need three things:

1. The exact definition of what colour should be displayed for value #007ba6. The definition is either implied (sRGB), or an explicit colour profile embedded in the image file. A common format for colour profiles is ICC. Other formats are available, but are not common.

2. Hardware (displays and printers) that respond predictably. This requires that the hardware is calibrated. It may involve the operating system, or a layer of software above the OS.

3. Display software (web browsers, photo viewers, etc) that respond correctly to embedded colour profiles, and convert image values from that profile to a profile suitable for the hardware.

You have no control of factors (2) and (3) of people who view your images. If you want the widest possible audience for your images, you should not assume their hardware is calibrated, nor that their hardware has a larger gamut than sRGB, nor that their software respects embedded colour profiles.

So I suggest you prepare your images in the sRGB colorspace. If that involves an embedded ICC profile, don't remove it.

As the years move on, perhaps monitors with gamuts larger than sRGB will become common.

Currently, images files are usually encoded as (red,green,blue) channels. Perhaps other colour models such as CIELab will become common in future.

**Comment by w4rner** [comment]
Can you clear up “ICC Profile” For me? Is that the same as ColorSync Profile?

If so, then given problems I’ve had with matching background color in Carrd website builder, it would seem I should reverse engineer the Carrd background to look the same as my Color-Sync-Profiled image.
(what i describe in my original post as "2. Figure out different hex value to put in Adobe to eventually look same as website”)

This seems perverse though. Surely a hex value on a webpage is “the true value” on the web? Is there any sense in which website backgrounds can have the same ColorSync Profile embedded in them to make them act the same way as the image?

W4RNER

**Reply by snibgo** [reply]
ColorSync Profiles are Apple's proprietary colour profiles. I have heard that a ColorSync Profile may be binary-identical to the equivalent ICC profile. I have no experience with ColorSync profiles.

I know nothing about "Carrd website builder".

Website colours in HTML or CSS are specified in the sRGB colorspace.

**Reply by w4rner** [reply]
Agreed: this Apple doc would seem to suggest that a ColorSync Profile is what they call a "[color profile](https://support.apple.com/en-gb/guide/colorsync-utility/csync005/mac)" more generally, which you can manipulate with their ColorSyncUtility.app
Confusingly an image can be in Color space RGB but then have a different Color profile.:
<img width="256" alt="image" src="https://user-images.githubusercontent.com/31451410/119648938-efb4de80-be19-11eb-8575-ff1c71d6e5c6.png">

---

In terms of websites, specific website builder irrelevant. The fact is in html code it specifies:
<img width="995" alt="image" src="https://user-images.githubusercontent.com/31451410/119649084-24289a80-be1a-11eb-9514-cbef269fc1a2.png">
And I'm wondering how to reliably choose color profiles which match that color (in sRGB to colorspace I guess!).

The unlikely conclusion I've come to is that only happens if you remove the sRGB color profile!

**Comment by JanCourtheyn** [comment]
I think I might have a similar issue.
Image rendition via digizuite (DAM software) vs local created image:
![image](https://user-images.githubusercontent.com/83012886/119525071-7fa54a80-bd7e-11eb-9e5d-c4020c78d1a8.png)
Left is local (and correct), right is via DAM software.

I use this code:
magick.exe convert "INPUT.PSD"[4] -profile sRGB.icc -strip -background none -resize 1900x1000 -gravity center -extent 1920x1080 -density 72x72 "OUTPUT.PNG"

Files can be found here: https://drive.google.com/drive/folders/1V41so2hyOgpHQbV-w4xVegfd5353at9b?usp=sharing

**Reply by fmw42** [reply]
In ImageMagick 7, use magick.exe not magick.exe convert.  Why do you add a profile and then strip it away? If you want it to be displayed consistently, keep the profile.

PNG uses density units of pixels per centimeter. So your print size may not be what you want.

**Reply by w4rner** [reply]
don't use Google so can't access files. 
But eyeballing: right looks brighter (but could just be because window is highlighted. Also as I discuss in original blog post, things can appear differently on other's screens when screenshotted. This is a seriously disconcerting topic for one's sense of consistent reality).

**Comment by snibgo** [comment]
I can't access those drive.google files. The right-hand image clearly has higher green saturation than the left image.

As fmw42 says, the embedded sRGB profile should not be stripped. But that should not create such a large difference in saturation.

Showing screenshots of images tells us nothing, because we don't know if the display software takes regard of embedded profiles, and many other factors.
